util/rlp/src/stream.rs

```diff
@@ -273,6 +317,17 @@ impl RlpStream {
 			false => panic!()
 		}
 	}
+
+	/// Finalize current ubnbound list. Panics if no unbounded list has been opened.
+	pub fn complete_unbounded_list(&mut self) {
+		let list = self.unfinished_lists.pop().expect("No open list.");
+		if list.max.is_some() {
+			panic!("List type mismatch.");
+		}
+		let len = self.buffer.len() - list.position;
+		self.encoder().insert_list_payload(len, list.position);
+		self.note_appended(1);
+	}
```

Comment:
It would only make sense if we forced each list to be a separate stream. What would it mean to call `begin_list(X)` on an `UnboundedRlpStream` or `begin_list_unbounded()` on a regular `Stream`?